### PR TITLE
Intergration/Cache/Coverage

### LINKED ISSUE #645: [CI] Cache dependencies source and build folders
CI fetch and recompile Radium dependencies at each runs.
As we now have a way to manage dependencies out of Radium, it could save a lot of time to cache the source and build directories of out-of-build dependencies. This would allow to:
 - save time by skipping compilation at each run
 - still allow dependencies updates by always calling the dependencies cmake project on the cached directories (should compile only in case of change).

### PR DESCRIPTION
**Pull request by dlyr**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature : add integration tests (that could not be done with unit tests only).
Example is given for tinyplyloader
Add also coverage computation (RADIUM_ENABLE_COVERAGE), and trigger codecov.io update action
Add external caches for github action.

* **What is the current behavior?** (You can also link to an open issue here)
no integration tests

* **What is the new behavior (if this is a feature change)?**
add cmake and dir structures to add integration tests with one example
add lcov-* targets

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
nope
Fixes #645 

* **Other information**:
no doc for the moment, and work on linux only I think, due to some scripts.

### CODE REVIEW

**Comment by nmellado** [comment]
Could you update GitHub Actions so they actually run the integration tests ?

**Comment by dlyr** [comment]
Well, they do already (since the check target trigger all ctest tests) see https://github.com/STORM-IRIT/Radium-Engine/pull/644/checks?check_run_id=1657452456
```
1/3 Test #1: catch_unit_tests .................   Passed    0.01 sec
    Start 2: tinyply-info_create
2/3 Test #2: tinyply-info_create ..............   Passed    0.07 sec
    Start 3: tinyply-info_compare
3/3 Test #3: tinyply-info_compare .............   Passed    0.00 sec
```

**Comment by nmellado** [comment]
I think that you need to reduce the number of compilation thread for the coverage build, the compiler crashed, probably running out of memory.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -17,41 +16,54 @@ else()
...
+    # TODO : find a way to that more efficiently
+    find_dependency(cpplocate REQUIRED NO_DEFAULT_PATH)
+    find_dependency(glbinding REQUIRED NO_DEFAULT_PATH)
+    find_dependency(globjects REQUIRED NO_DEFAULT_PATH)
+    find_dependency(tinyply REQUIRED NO_DEFAULT_PATH)
+    set(RadiumExternalDlls_location "")
+    addImportedDir( FROM cpplocate::cpplocate TO RadiumExternalDlls_location) 
+    addImportedDir( FROM glbinding::glbinding TO RadiumExternalDlls_location) 
+    addImportedDir( FROM globjects::globjects TO RadiumExternalDlls_location)
+    addImportedDir( FROM tinyply TO RadiumExternalDlls_location)
```
Comment (nmellado) [inline tests/CMakeLists.txt:31]: The use of tinyply is garded by the option `RADIUM_IO_TINYPLY`, so I suggest:
```suggestion
    if (RADIUM_IO_TINYPLY)
        addImportedDir( FROM tinyply TO RadiumExternalDlls_location) 
    endif ()
```

Comment (dlyr) [reply tests/CMakeLists.txt]: Should be done. If I didn't mixup with rebases.

**Comment by nmellado** [comment]
Could we mention in the documentation that submodules are required (changes the compilation instructions)

**Comment by dlyr** [comment]
Well in fact I have second thought, I will remove the submodule

**Comment by dlyr** [comment]
There is an inconsistency when I run from my own install and when it runs from CI, the report is not the same.
Let's do some tests before accepting the PR.

**Review by nmellado — APPROVED** [review]

**Review by nmellado — APPROVED** [review]

**Pull request merged**